src/osvr/Common/GestureComponent.cpp

```diff
@@ -0,0 +1,236 @@
+                                 OSVR_GestureID gestureID,
+                                 OSVR_ChannelCount sensor)
+                : m_gestureState(state), m_gestureID(gestureID),
+                  m_sensor(sensor) {}
+
+            MessageSerialization() {}
+
+            template <typename T> void processMessage(T &p) {
+                p(m_gestureState);
+                p(m_gestureID);
+                p(m_sensor);
+            }
+            GestureData getData() const {
+                GestureData ret;
+                ret.sensor = m_sensor;
+                ret.gestureID = m_gestureID;
+                ret.gestureState = m_gestureState;
+                return ret;
+            }
+
+          private:
+            OSVR_GestureState m_gestureState;
+            StringID m_gestureID;
+            OSVR_ChannelCount m_sensor;
+        };
+        const char *GestureRecord::identifier() {
+            return "com.osvr.gesture.gesturerecord";
+        }
+
+        class GestureMapRecord::MessageSerialization {
+          public:
+            MessageSerialization(SerializedStringMap serializedMap)
+                : m_serializedMap(serializedMap) {}
+
+            MessageSerialization() {}
+
+            template <typename T> void processMessage(T &p) {
+                p(m_serializedMap);
+            }
+            GestureMap getData() const {
+                GestureMap ret;
+                ret.serializedMap = m_serializedMap;
+                return ret;
+            }
+
+          private:
+            SerializedStringMap m_serializedMap;
+        };
+        const char *GestureMapRecord::identifier() {
+            return "com.osvr.gesture.gesturemaprecord";
+        }
+
+    } // namespace messages
+
+    shared_ptr<GestureComponent>
+    GestureComponent::create(OSVR_ChannelCount numChan) {
+        shared_ptr<GestureComponent> ret(new GestureComponent(numChan));
+        return ret;
+    }
+
+    GestureComponent::GestureComponent(OSVR_ChannelCount numChan)
+        : m_numSensor(numChan) {
+
+        // populate the gesture map
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_SWIP_LEFT);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_SWIP_RIGHT);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_SCROLL_UP);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_SCROLL_DOWN);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_SINGLE_TAP);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_DOUBLE_TAP);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_PINCH);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_FINGER_SPREAD);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_CIRCLE);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_LONG_PRESS);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_OPEN_HAND);
+        m_gestureNameMap.registerStringID(OSVR_GESTURE_CLOSED_HAND);
+    }
+
+    void GestureComponent::recordParent(Parent &dev) {
```

Comment:
This is the purpose of having `m_parentSet()` - so you don't override this method if you want to hook in your behavior.